### PR TITLE
checkbox - new size property, css cleanup, demo and readme updates

### PR DESCRIPTION
**Pull request by vesper2000**
# patch

### CODE REVIEW

File: addon/components/frost-checkbox.js [hunk 1]
```diff
@@ -5,10 +5,24 @@ import _ from 'lodash/lodash'
...
   layout: layout,
   classNames: ['frost-checkbox'],
+  classNameBindings: ['size'],
 
   didInitAttrs () {
     const checked = this.attrs ? this.attrs.checked : false
     this.set('isChecked', checked)
+
+    const checkboxSize = this.get('size')
+    switch (checkboxSize) {
```
Comment (sglanzer-fire) [inline addon/components/frost-checkbox.js:15]: Is this whole switch necessary?  Seems like it's mapping a property to itself

File: tests/dummy/app/pods/checkbox/template.hbs [hunk 2]
```diff
@@ -1,172 +1,154 @@
...
-{{frost-checkbox id='basic'}}
-```"
-}}
+<div class="dummy-notification-container">
+{{#each notifications as |notification|}}
+  {{notification-message notification=notification}}
+{{/each}}
+</div>
+
+<div class='dummy-body text-input field'>
```
Comment (sglanzer-fire) [inline tests/dummy/app/pods/checkbox/template.hbs]: Why are we using the `text-input` `field` classes here?

File: tests/dummy/app/pods/checkbox/template.hbs [hunk 3]
```diff
@@ -1,172 +1,154 @@
...
+```"}}
+      </div>
     </div>
-    <div class="demo">
-      {{frost-checkbox id="basic"}}
+
+   <div class='example'>
+      <div class='title'>Size</div>
+      <div class='demo'>
+        {{frost-checkbox size='large'}}
```
Comment (sglanzer-fire) [inline tests/dummy/app/pods/checkbox/template.hbs:56]: Maybe small/medium/large side by side here?

File: tests/dummy/app/pods/checkbox/template.hbs [hunk 4]
```diff
@@ -1,172 +1,154 @@
...
-  </div>
 
-  <div class="example">
-    <div class="title">Checked</div>
-    <div class="snippet">
-{{format-markdown "```handlebars
+    <div class='example'>
+      <div class='title'>Checked</div>
+      <div class='demo'>
+        {{frost-checkbox checked='true'}}
```
Comment (sglanzer-fire) [inline tests/dummy/app/pods/checkbox/template.hbs]: Is it text 'true' or `true` (boolean)?

File: tests/dummy/app/pods/checkbox/template.hbs (hunk 2 above)
Comment (rox163) [inline tests/dummy/app/pods/checkbox/template.hbs]: @vesper2000 the t'ext-input field' class defines a width on a per component basis to help set a max width for all the snippet blocks. So you can add one for checkbox based on what a good width would be for them

Comment (vesper2000) [inline tests/dummy/app/pods/checkbox/template.hbs]: Cool - will do that.

File: tests/dummy/app/pods/checkbox/template.hbs (hunk 4 above)
Comment (vesper2000) [inline tests/dummy/app/pods/checkbox/template.hbs]: It is a boolean, added '' to make it standout, but I guess that is not correct - the readme has it as boolean.

File: tests/dummy/app/pods/checkbox/template.hbs (hunk 3 above)
Comment (vesper2000) [inline tests/dummy/app/pods/checkbox/template.hbs:56]: I had that and then I removed it...  I figured it was just an example and the available options are in the readme.

File: tests/dummy/app/pods/checkbox/template.hbs [hunk 5]
```diff
@@ -1,172 +1,154 @@
...
-  </div>
 
-  <div class="example">
-    <div class="title">Name</div>
-    <div class="snippet">
-{{format-markdown "```handlebars
+    <div class='example'>
+      <div class='title'>Disabled - not checked</div>
+      <div class='demo'>
+        {{frost-checkbox disabled='true'}}
```
Comment (sglanzer-fire) [inline tests/dummy/app/pods/checkbox/template.hbs]: Same deal about text vs. boolean

File: addon/components/frost-checkbox.js (hunk 1 above)
Comment (vesper2000) [inline addon/components/frost-checkbox.js:15]: No necessary, but I did it this way to catch the case when no size is specified.

File: tests/dummy/app/pods/checkbox/template.hbs (hunk 5 above)
Comment (vesper2000) [inline tests/dummy/app/pods/checkbox/template.hbs]: Yeah - I will update all of them.

File: tests/dummy/app/pods/checkbox/template.hbs (hunk 3 above)
Comment (sglanzer-fire) [inline tests/dummy/app/pods/checkbox/template.hbs:56]: I think it might look nice?  If it doesn't that's fine

**Comment by sglanzer-fire** [comment]
Approved - minor tweaks as noted

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-core/pull-request/13/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-core)

File: tests/dummy/app/pods/checkbox/template.hbs (hunk 3 above)
Comment (vesper2000) [inline tests/dummy/app/pods/checkbox/template.hbs:56]: Ok - added it in :)

**Comment by vesper2000** [comment]
@sglanzer if you can re-approve please - the last commit un-did the approval - tx!

**Comment by sglanzer-fire** [comment]
Approved

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-core/pull-request/13/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-core)

**Pull request merged**